dapp/src/components/TransactionListener.js

```diff
@@ -244,57 +244,64 @@ class TransactionListener extends Component {
     }
   }
 
-  async observeTransactions(transactionsToCheck) {
-    const nonMinedTx = transactionsToCheck
-      .filter((t) => !t.mined && !t.observed)
-      .map((t) => {
-        const newTx = { ...t }
-        newTx.observed = true
-        return newTx
+  const observeTransactions = async (transactionsToCheck) => {
+    const resolvedTransactions = await Promise.all(
+      transactionsToCheck.map(async (t) => {
+        if (!t.isSafe) return t
+        // If this was a Gnosis Safe transaction, we have a SafeTxHash not a
+        // real transaction hash. We need to resolve some additional data about
+        // the transaction using the Gnosis Safe SDK.
+        let safeData
+        // TODO handle a 404 here. We need to retry. A refresh is required to
+        // get this to retry at the moment.
+        try {
+          safeData = await connector.sdk.txs.getBySafeTxHash(t.hash)
+        } catch (e) {
+          console.error(e)
```

Comment:
nit: Don't know if this error is going to be descriptive, some context might help: 
```
console.error('Gnosis safe SDK call failed: ', e)
```